class.jetpack.php

```diff
@@ -7280,6 +7285,38 @@ public static function is_rewind_enabled() {
 		return $rewind_enabled;
 	}
 
+	/**
+	 * Checks for whether Jetpack Scan is enabled.
+	 * Will return true if the state of Scan is anything except "unavailable".
+	 *
+	 * @return bool|int|mixed
+	 */
+	public static function is_scan_enabled() {
+		if ( ! self::is_active() ) {
+			return false;
+		}
+
+		$scan_enabled = get_transient( 'jetpack_scan_enabled' );
+		if ( false === $scan_enabled ) {
+			jetpack_require_lib( 'class.core-rest-api-endpoints' );
```

Comment:
Instead of doing this on page load every 10 minutes. I think we should do think in async fashion. In a cron job or on the front end using js. This way the loading of pages that show the admin bar will not be delayed by the server request.